model/src/main/kotlin/FileFormat.kt

```diff
@@ -94,9 +94,14 @@ fun File.mapper() = FileFormat.forFile(this).mapper
 fun File.readTree(): JsonNode = mapper().readTree(this)
 
 /**
- * Use the Jackson mapper returned from [File.mapper] to read an object of type [T] from this file.
+ * Use the Jackson mapper returned from [File.mapper] to read a single object of type [T] from this file.
  */
-inline fun <reified T : Any> File.readValue(): T = mapper().readValue(this)
+inline fun <reified T : Any> File.readValue(): T {
+    val mapper = mapper()
+    val parser = mapper.factory.createParser(this)
+    return mapper.readValues<T>(parser).readAll().singleOrNull()
+        ?: throw IllegalArgumentException("Multiple documents found in file '$this' which is not supported.")
+}
```

Comment:
Should a similar logic be applied to `readValueOrNull()` or `readValueOrDefault()` ?

Reply:
What would be your expectation for `readValueOrNull()` in case of multiple documents per file, should it return `null` or throw?

Reply:
I went for throwing, just like for invalid content.